fe/fe-core/src/main/java/com/starrocks/failpoint/README.md

```diff
@@ -0,0 +1,56 @@
+# What is failpoint?
+Failpoint is a fault injection testing framework that can precisely inject faults at any location within a function, helping to test system fault tolerance and stability.  
+Writing failpoints does not require modifying the source code of the system under test (SUT), making it suitable for both R&D engineers (RD) and quality assurance (QA).
+
+# How to use failpoint?
+## Defining a failpoint
+Defining a failpoint is very simple. Here's an example:  
+
+```text
+RULE bdb_ha_get_leader_exception
+CLASS com.starrocks.ha.BDBHA
+METHOD getLeader()
+HELPER com.starrocks.failpoint.FailPointHelper
+IF shouldTrigger("bdb_ha_get_leader_exception")
+DO throw new RuntimeException("failpoint triggered");
+ENDRULE
+```
+
+This is a failpoint written in Byteman script that triggers an exception when calling `com.starrocks.ha.BDBHA.getLeader`.
+
+- `RULE`: The name of the rule.
+- `CLASS`: The class name where the fault is injected.
+- `METHOD`: The method name where the fault is injected. You can also specify parameter types for overloaded methods, e.g., `getLeader(int)`.
+    - Below `METHOD`, you can define the injection location, such as:
+        - `AT ENTRY`: At the beginning of the method.
+        - `AT EXIT`: At the end of the method.
```

Comment:
what if i want to break the logic in the middle, how should i achieve that?

Reply:
![image](https://github.com/user-attachments/assets/131fe20a-99f0-43d6-9ea5-783b4fff27e2)
You can inject failure points when/after certain variables are read/written, or when/after certain methods are called.